beacon-chain/core/state/epoch_processing.go

```diff
@@ -0,0 +1,64 @@
+package state
+
+import (
+	"bytes"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/types"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// EpochAttestations returns the pending attestations of slots in the epoch
+// (state.slot-EPOCH_LENGTH...state.slot-1), not attestations that got
+// included in the chain during the epoch.
+//
+// Spec pseudocode definition:
+//   this_epoch_attestations = [a for a in state.latest_attestations
+//   if state.slot - EPOCH_LENGTH <= a.data.slot < state.slot]
+func EpochAttestations(state *pb.BeaconState) []*pb.PendingAttestationRecord {
+	epochLength := params.BeaconConfig().EpochLength
+	var thisEpochAttestations []*pb.PendingAttestationRecord
+	var earliestSlot uint64
+
+	for _, attestation := range state.LatestAttestations {
+
+		// If the state slot is less than epochLength, then the earliestSlot would
+		// result in a negative number. Therefore we should default to
+		// earliestSlot = 0 in this case.
+		if state.Slot > epochLength {
+			earliestSlot = state.Slot - epochLength
+		}
+
+		if earliestSlot <= attestation.Data.Slot && attestation.Data.Slot < state.Slot {
```

Comment:
Use the getters here when nesting dot properties of proto structs. For example, prefer `attestation.GetData().GetSlot()` as that will avoid any panics if data is undefined. In the future, we can replace this with our own `*types.Attestation` once we complete the refactor.